crates/ruma-common/src/identifiers/key_id.rs

```diff
@@ -12,6 +12,36 @@ use super::{
 };
 
 /// A key algorithm and key name delimited by a colon.
+///
+/// Examples of the use of this struct are [`DeviceKeyId`], which identifies a Ed25519 or Curve25519
+/// [device key](https://spec.matrix.org/v1.12/client-server-api/#device-keys), and
+/// [`CrossSigningKeyId`], which identifies a user's [cross signing key](https://spec.matrix.org/v1.12/client-server-api/#cross-signing).
+///
+/// This format of identifier is often used in the `signatures` field of [signed JSON](https://spec.matrix.org/v1.12/appendices/#signing-details)
+/// where it is referred to as a "signing key identifier".
+///
+/// This struct is rarely used directly - instead you should expect to use one of the typedefs
+/// that rely on it like [`CrossSigningKeyId`] or [`DeviceSigningKeyId`].
+///
+/// # Examples
+///
+/// To parse a colon-separated identifier:
+///
+/// ```rust
+/// use ruma_common::{DeviceId, DeviceKeyAlgorithm, KeyId, OwnedKeyId};
```

Comment:
nit: Let's format the examples like rustfmt would format the code. Here it means that there should be an empty line between the `use` declaration and the rest of the code.

If we tell the users to use the type aliases, wouldn't it make more sense to use one of them here?

Reply:
Thanks, fixed in c77d2e82